internal/service/enrollmentrequest.go

```diff
@@ -383,24 +384,31 @@ func (h *ServiceHandler) ApproveEnrollmentRequest(ctx context.Context, request s
 		if v1alpha1.IsStatusConditionTrue(enrollmentReq.Status.Conditions, v1alpha1.EnrollmentRequestApproved) {
 			return server.ApproveEnrollmentRequest400JSONResponse{Message: "Enrollment request is already approved"}, nil
 		}
-		if request.Body.ApprovedAt != nil {
-			return server.ApproveEnrollmentRequest400JSONResponse{Message: "ApprovedAt is not allowed to be set when approving enrollment requests"}, nil
+
+		identity, err := authcommon.GetIdentity(ctx)
+		if err != nil {
+			return server.ApproveEnrollmentRequest500JSONResponse{Message: fmt.Sprintf("Failed to retrieve user identity while approving enrollment request: %v", err.Error())}, nil
```

Comment:
If you return the error as second parameter (instead of `nil`), this'll generate a 500 error.

Reply:
great, thanks for pointing that out! Fixed.